### DISCUSSION TITLE
Unraid Plugin

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#455] [General]
@ich777

I'm unfortunately oOo with limited access. My advices for below issues

# Power consumption
This Unraid [issue](https://forums.unraid.net/topic/109314-plugin-corefreq/?do=findComment&comment=1277087)
Don't make a confusion with measured watt on the wall. CoreFreq is supplying the [RAPL](https://www.google.com/search?q=Intel+rapl&oq=Intel+rapl&aqs=chrome..69i57j69i65l2.6624j0j7&client=ms-android-huawei&sourceid=chrome-mobile&ie=UTF-8#ip=1) estimation. Thus User is getting parts of the whole system. Per Processor counters with Intel ; both per Core and per Processor for AMDs

# Turbo
This [issue](https://forums.unraid.net/topic/109314-plugin-corefreq/?do=findComment&comment=1292675)
What User is achieving is to stress Core(s) to trigger the Turbo frequencies. See Tools as an instant mean for Turbo verification. 
To set Turbo and other linked settings, parameters have to be defined. 
```
modinfo corefreqk.ko
```
will supply the full list of accepted parameters (for a GUI help) to store with User value in the usual `modprobe.conf` location 
Be sure to offer User a mean to cancel boot parameters in case of unsupported values. 
```
TurboBoost_Enable=1
```
will activate the Turbo in Processor and select an adequate Target frequency ratio (depending on the architecture)

**Comment by ich777** [comment]
Thanks for the response @cyring, so to speak this user has to only use: `TurboBoost_Enable=1` since it seems to be disable by default, correct?

I've now also created a Module options (directly read from `modinfo`) page where it is described how to make changes persistent, hope that's appropriate:
![grafik](https://github.com/cyring/CoreFreq/assets/28066518/b2358d43-6bb3-477d-ac5d-106078add738)

I'll wait with the release until your response if this is okay.

**Comment by cyring** [comment]
> Thanks for the response @cyring, so to speak this user has to only use: `TurboBoost_Enable=1` since it seems to be disable by default, correct?

I have understood the same as you. User just needs this  parameter. 


> 
> I've now also created a Module options (directly read from `modinfo`) page where it is described how to make changes persistent, hope that's appropriate: ![grafik](https://user-images.githubusercontent.com/28066518/259390886-b2358d43-6bb3-477d-ac5d-106078add738.png)
> 
> I'll wait with the release until your response if this is okay.

Looks good and that's the way to do it. 

However some parameters are really hazardous : `ArchID` is an Architecture selector, mainly for debugging purpose and can lead to a crash if wrongly choosen. Don't list this parameter. 
`Experimental` is to debug code features also.
`CPU_Count` will override the monitoring scale but harmless.
The remaining can be listed.

**Reply by ich777** [reply]
@cyring I can of course hide them but I would like to not do that because it's up to the users to configure it properly and I've even placed the WARNING in there so that users know what to do.

**Reply by cyring** [reply]
Edit: also some parameters have to be set coherently.
```
SpeedStep_Enable=1 TurboBoost_Enable=1
```
because Intel Turbo works that way.

However these bits are too long to explain in one single help page. 
User has to observe in the UI how features are tight together to claim the same with parameters. 
And should read specifications of the underlying technologies. 

_let me know if my english explanation does not make sens_

**Comment by cyring** [comment]
> @cyring I can of course hide them but I would like to not do that because it's up to the users to configure it properly and I've even placed the WARNING in there so that users know what to do.

I do agree.

**Comment by ich777** [comment]
Thanks for the heads up about the examples, I've already pushed the update but I will change the examples in the next update, maybe tomorrow.
Hope that explains it a bit better for Unraid users.

As always, thanks for the help!
